bundle/parallel_test.go

```diff
@@ -10,35 +12,58 @@ import (
 )
 
 type addToContainer struct {
+	t         *testing.T
 	container *[]int
 	value     int
 	err       bool
+
+	// mu is a mutex that protects container. It is used to ensure that the
+	// container slice is only modified by one goroutine at a time.
+	mu *sync.Mutex
 }
 
 func (m *addToContainer) Apply(ctx context.Context, b ReadOnlyBundle) diag.Diagnostics {
 	if m.err {
 		return diag.Errorf("error")
 	}
 
-	c := *m.container
-	c = append(c, m.value)
-	*m.container = c
+	m.mu.Lock()
+	*m.container = append(*m.container, m.value)
+	m.mu.Unlock()
+
 	return nil
 }
 
 func (m *addToContainer) Name() string {
 	return "addToContainer"
 }
 
-func TestParallelMutatorWork(t *testing.T) {
+func TestParallelMutatorWork1000Times(t *testing.T) {
+	for i := 0; i < 1000; i++ {
+		t.Run(fmt.Sprintf("iter %v", i), func(t *testing.T) {
+			testParallelMutatorWork(t)
+		})
+	}
+}
+
+func TestParallelMutatorWorkWithErrors1000Times(t *testing.T) {
+	for i := 0; i < 1000; i++ {
+		t.Run(fmt.Sprintf("iter %v", i), func(t *testing.T) {
+			testParallelMutatorWorkWithErrors(t)
+		})
+	}
+}
```

Comment:
Can these be removed now that the fix is in?

You can run the same test N times by specifying the `-test.count` flag I believe.

Reply:
Removed. Added that to prevent a regress but it's fairly unlikely to regress. At some point (maybe) we can add a flakyness detected for new tests added to the repo. Like, running a new unit test 100 times or new integration tests 10 times to ensure they are not flaky. 

Not necessary ATM since flakyness has not been an issue.